python/llm/src/bigdl/llm/transformers/models/llama.py

```diff
@@ -522,6 +522,10 @@ def check_flash_attention_available(query):
     if query.dtype not in [torch.float32, torch.float16]:
         # only use flash attention for fp32/fp16 input
         return False
+    if query.shape[0] > 1:
+        # only use flash attention for batch_size = 1
+        # as we found some precision problem with batch_size > 1
+        return False
     return True
```

Comment:
Maybe change to:
```python
def use_flash_attention(query):
   if query.size()>1 || query.shape[0] > 1:
       return false;
  ...

```

Reply:
Done.

Reply:
Move it to the beginning of the method

Reply:
Not quite understanding about "remove change q_len>1 in the forward method"?

Reply:
You are already checking query length here, and [q_len>1](https://github.com/intel-analytics/BigDL/blob/main/python/llm/src/bigdl/llm/transformers/models/llama.py#L143) is no longer need in the forward method

Reply:
I don't check query length here, `if query.shape[0] > 1 or query.size()[0] > 1:` only check batch size of query (query shape is [B, S, H] here).
But I can move `q_len > 1` into check_flash_attention_available(query) .

Reply:
`bsz, q_len, _ = hidden_states.size()`, yes?

Reply:
Yes.